text/0000-idempotent-install.md

```diff
@@ -0,0 +1,68 @@
+- Start Date: 2016-12-13
+- RFC PR: (leave this empty)
+- Yarn Issue: (leave this empty)
+
+# Summary
+
+`yarn install` should be idempotent.
+
+Ideally, the result of `yarn install` would not be statefully dependent on the contents of an existing `node_modules`, and would ensure the resulting `node_modules` is identical regardless of whether there is an existing `node_modules` or not. This seems very inline with the spirit of yarn's goal of deterministic builds (same `node_modules` independent of whether `node_modules` exists or the version of node that generated it).

```

Comment:
I'm not sure I understand. It sounds like you're saying we should add the functionality of checking the node version, and then asking how that's different than the current behavior?

FWICT, yarn does not currently handle the issue as described:

```
[12:07:43] /tmp/foo (master)  $ ls
package.json
[12:07:50] /tmp/foo (master)  $ cat package.json
{
  "name": "foo",
  "dependencies": {
    "node-sass": "^4.0.0"
  }
}
[12:07:54] /tmp/foo (master)  $ node -v
v4.7.0
[12:07:57] /tmp/foo (master)  $ yarn add node-sass >/dev/null 2>&1
[12:08:19] /tmp/foo (master)  $ ls
node_modules/ package.json  yarn.lock
[12:08:21] /tmp/foo (master)  $ node -e "require('node-sass')"
[12:08:22] /tmp/foo (master)  $ yarn check --integrity
yarn check v0.18.0
warning foo: No license field
success Folder in sync.
✨  Done in 0.49s.
[12:08:28] /tmp/foo (master)  $ nvm use v6
Now using node v6.9.2 (npm v3.10.10)
[12:08:33] /tmp/foo (master)  $ yarn check --integrity
yarn check v0.18.0
warning foo: No license field
success Folder in sync.
✨  Done in 0.44s.
[12:08:39] /tmp/foo (master)  $ yarn install
yarn install v0.18.0
warning foo: No license field
[1/4] 🔍  Resolving packages...
success Already up-to-date.
✨  Done in 0.14s.
[12:08:44] /tmp/foo (master)  $ node -e "require('node-sass')"
/private/tmp/foo/node_modules/node-sass/lib/binding.js:15
      throw new Error(errors.missingBinary());
      ^

Error: Missing binding /private/tmp/foo/node_modules/node-sass/vendor/darwin-x64-48/binding.node
Node Sass could not find a binding for your current environment: OS X 64-bit with Node.js 6.x

Found bindings for the following environments:
  - OS X 64-bit with Node.js 4.x

This usually happens because your environment has changed since running `npm install`.
Run `npm rebuild node-sass` to build the binding for your current environment.
    at module.exports (/private/tmp/foo/node_modules/node-sass/lib/binding.js:15:13)
    at Object.<anonymous> (/private/tmp/foo/node_modules/node-sass/lib/index.js:14:35)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at [eval]:1:1
```